### ISSUE TITLE
(PHP) Syntax parsing breaks inside parameters of some function calls

### ISSUE THREAD

**Issue by Roy-Orbison** [highlightjs/highlight.js#3655]
**Describe the issue**
When a lambda function is used as a parameter, it breaks down if it has a `use` section after its parameters.

Comments break inside object method calls (but apparently not regular function calls).

**Which language seems to have the issue?**
PHP

**Are you using `highlight` or `highlightAuto`?**
I noticed it on Stack Overflow, and confirmed with `highlightElement` in your demo pages.

**Sample Code to Reproduce**
![As highlighted on your demo page](https://user-images.githubusercontent.com/28617290/200727770-3e79ecbc-ac70-4be8-99d5-9ce12e33bbcd.png)

**Expected behavior**
```php
funky(
	'string',
	function(array $x) use ($y) {
		# $y above fails to parse as a var, and this comment is broken
		$x[] = $y
		return $x;
	}
);

$o->m([
	'foo',
	# this comment highlights keywords like include and 'string' literals
	'bar',
]);
```

**Comment by wkania** [comment]
Yes the comments are somehow wrongly parsed. I noticed it on ChatGPT [answer](https://highlightjs.org/demo#lang=php&v=1&theme=atom-one-dark&code=dXNlIFN5bWZvbnlcQ29tcG9uZW50XEV2ZW50RGlzcGF0Y2hlcsYQU3Vic2NyaWJlckludGVyZmFjZTsK1kBIdHRwS2VybmVsXMYHxUFz2i9tbFNhbml0aXplcs4Oy2sKcmVhZG9ubHkgY2xhc3MgRmlsZVVwbG9hZMkuTGlzdGVuZXIgaW1wbGVtxHQg%2BAC8CnsKICAgIHB1YmxpYyBmdW5jdGlvbiBfX2NvbnN0cnVjdCjFIcUlcml2YXRlIPYAmyAkc8gUIC8v6ADgJ3Mgyhdmb3Igc2FmZSBmaWxlbmFtZXPFXCkg5gCFfQrsAIxzdGF07ACTZ2XqAMBk5gFKKCk6IGFycmF5xTjGRcQBcmV0dXJuIFvJEcQB7AGDOjpSRVFVRVNUID0%2BICfoAKHkAUtO5ACWJyzJOl07xQt9Cn0K).

Will look at it.

**Comment by hashar** [comment]
With #4230 I am fixing the C line comments (`// comment`). But I guess the patch should also fixes the comments starting with `#` :)

**Comment by joshgoebel** [comment]
Yes, let us know if #4230 does the trick.

**Comment by Roy-Orbison** [comment]
Do you have a sandbox to test in? If not, can you just paste the snippet above into your build to see if it's fixed?

**Comment by hashar** [comment]
> With [#4230](https://github.com/highlightjs/highlight.js/pull/4230) I am fixing the C line comments (`// comment`). But I guess the patch should also fixes the comments starting with `#` :)

After I have left that comment above, I have amended my pull request to add support for hash comments. The diff can be seen at https://github.com/highlightjs/highlight.js/compare/80f3fa7bee289cbdb21277da1f0e0e216559d6bd..6060d5c68794ed43f2e036d0607404b7e33264b8 and it adds a test to validate that hash comment contained within a function is rendered as a comment span :)
